### ISSUE TITLE
Make it easier to understand who enemies are targeting

### ISSUE THREAD

**Issue by TheInternetIsReal** [Luxocracy/NeatPlates#153]
HI, love NeatPlates. I was wondering if either of these are possible for display options:

**1.** Show the target class icon/playername on the bar of enemies. This would make healing DPS that pull aggro much easier.
**2.** Color the health plate yellow (or whatever) if the enemies are not attacking the tank OR you (so party members). This is more of a fallback if 1 can't be done but I could see it being nice to have with 1 as well.
     **2a.** Create a NeatPlates minimap button and enable it to set the designated tank when left clicking the button and targeting a player.

For the life of me I could not figure out how you are determining if an enemy is attacking the player or not and then turning that health bar red in classic. Therefore I can't tell how difficult my asks are.

**Comment by Luxocracy** [comment]
1. This can already be done by changing the _Enemy Status Text_ to _Target Of_, but it does replace any other information you might want there such as health. So I know it's not ideal, and the status text system is something that I've wanted to improve for a while but haven't had time/gotten around to as it's a major change.

2. While I do like the idea of this, the issue is really that there isn't a reliable way to get who is the tank in classic. And while your suggestion in **2a**, is a potential solution it's not very elegant and requires an entirely new system just to change, keep track of, verify the existence of that tank, and display this to the player in some way. As this simple _point & click_ system wouldn't really work too well in a raid environment where you have multiple tanks.
It just seems like a lot of work for something very few people would actually use. If it was automated I could see the value, but other than that it just seems like a major headache on my part.

There was (well it still exists technically) a system for something similar to this that highlighted the friendly nameplate of the unit that had aggro, which I feel like is something you really would have liked. Until people started exploiting friendly nameplates and blizzard disabled the modification of it in dungeons/raids.

I know there was also some unit frame addons that highlighted the party frame if the unit had aggro, but I'm not sure if these would work in Classic.

**Comment by TheInternetIsReal** [comment]
I had no idea 1 was already possible!! This is essentially the functionality I was looking for. 

If you need any ideas to improve it.... I'd prefer an option for where the anchor the target name  relative to the actual health bar. I'd put the name below the health bar. I'd also like to see the names colored by class which would just be an amazing brain processing shortcut for realizing who is targeted.

**Comment by Luxocracy** [comment]
Not sure why I haven't responded to this, maybe I just missed it, so first I want to apologize for that.

Secondly, as for the name of the option. Just calling it "Target" seems wrong to me as I've always known this sort of feature as "Target of Target", but in this case it's the target of that nameplate.

Changing the anchoring point & position of the status text is something that I am working on, but for pretty much every element of the nameplate, as moving various elements is something that is requested quite often. And there is an early **Alpha** release of this, though it still needs some work.

A colored version of "Target Of" exists, however I believe it was added around the time I last posted in this issue. So it might have been added after, but at any rate it exists.

And lastly, the "daunting" scrolled list of options is something I'm aware of and I think can be done better, but I also know that there are people who prefer it the way that it currently is. So I haven't been willing to spend time on changing it.

Now for what I really came here to mention was that I added a new slash command in the latest release(**/nptank**) that lets you assign tanks manually. As this is something that has been requested a few times since you mentioned something similar. And I think I was just overthinking the implementation of it when you brought it up last time, as this current implementation is rather simple.

Although to make it do what I think you were asking for you would have to set your own _Threat Mode_ to **Tank**, enable _Highlight Mobs on Off-Tanks_, and then change the _Safe_ color to the same as _Warning_. As _Safe_ would mean that the unit is targeting you.

I could potentially add another threat mode specifically for healers that basically does what I wrote above, but without having to mess with the predefined colors if that is something that you are interested in.
